app/src/main/res/values/styles.xml

```diff
@@ -465,10 +465,11 @@
     <item name="android:gravity">center_horizontal</item>
   </style>
 
-  <!-- ShapeableImageView -->
-  <!-- context for ShapeableimageView https://howtodoandroid.com/shapeableimageview-material-components-android/
-       -->
-  <!-- Here key to the overlay behavior is the empty parent attribute. This is what causes the overlay behavior -->
+  <!-- ShapeableImageView
+     Note that When defining a style to be used as a shapeAppearanceOverlay in shapeableImageView , we generally
+     set an empty parent by setting parent="". This is what causes the overlay behavior and also
+     shapeAppearanceOverlay stacks on top of shapeAppearance, so all values will be set in the shapeAppearance.
+     context for ShapeableImageView https://howtodoandroid.com/shapeableimageview-material-components-android/ -->
```

Comment:
This still doesn't quite explain _why_ the parent bit works--that's the thing that's important to include, I think, since otherwise we won't understand how to debug this if it breaks in the future (or doesn't work for some specific situation).

Reply:
I didn't get what exactly I should write , I found this much only in the documents related to ShapeableImageView regarding this empty parent attribute.

Reply:
Sorry, should've followed up on this earlier. We generally shouldn't be making changes just because they work--it's really important to understand the 'why' (otherwise we may unintentionally bring extra consequences that we might miss adopting a solution that may only seem correct on the surface). I think we need to better understand why this works & reflect that here so that there isn't confusion.

I think the main gap is understanding why setting an empty parent works specifically, since that's actually rather surprising. I'm assuming it's preventing some sort of override that would otherwise prevent this behavior (but that would only be true if an overlay is the default, which I'm not sure). This uncertainty is actually I think it's important to fully understand the solution.

Reply:
The latest wording addresses my concern on clarity (mostly because the latest code is more self-documenting).